System.Threading.Thread tests crashing on Mono

https://dev.azure.com/dnceng/public/_build/results?buildId=1037924&view=ms.vss-test-web.build-test-results-tab&runId=32176598&paneView=attachments&resultId=180347
https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-49582-merge-946cc614d5d64401bf/System.Threading.Thread.Tests/console.348ec2f2.log?sv=2019-07-07&se=2021-04-02T17%3A22%3A03Z&sr=c&sp=rl&sig=y5h54yQ5kT6ij4I3eLYK4J2iA8kFcMStqR5%2Bglu45Vk%3D
```
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
   at System.Threading.WaitSubsystem.OnThreadExiting(Thread thread) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs:line 430
   at System.Threading.Thread.Finalize() in /_/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs:line 85
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
   at System.Threading.WaitSubsystem.OnThreadExiting(Thread thread) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs:line 430
   at System.Threading.Thread.Finalize() in /_/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs:line 85
```

Seen in https://github.com/dotnet/runtime/pull/49608 `runtime (Libraries Test Run release mono OSX x64 Debug)`

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-49608-merge-452b8f6c8f6b417182/System.Threading.Thread.Tests/console.11032fa9.log?sv=2019-07-07&se=2021-04-03T23%3A52%3A11Z&sr=c&sp=rl&sig=aRKn6vyA2vMqWsnV9sa22QB22zjiuAlILDvd6fJ0npU%3D
````
=== TEST EXECUTION SUMMARY ===
   System.Threading.Thread.Tests  Total: 53, Errors: 0, Failed: 0, Skipped: 2, Time: 2.894s

Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
   at System.Threading.WaitSubsystem.OnThreadExiting(Thread thread) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs:line 430
   at System.Threading.Thread.Finalize() in /_/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs:line 85
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
   at System.Threading.WaitSubsystem.OnThreadExiting(Thread thread) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs:line 430
   at System.Threading.Thread.Finalize() in /_/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs:line 85
/private/tmp/helix/working/A8F30958/w/B84F09F9/e
````

--- a/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.Unix.cs
@@ -424,10 +424,5 @@ namespace System.Threading
                 s_lock.Release();
             }
         }
-
-        public static void OnThreadExiting(Thread thread)
-        {
-            thread.WaitInfo.OnThreadExiting();
-        }
     }
 }

--- a/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs
+++ b/src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs
@@ -82,7 +82,7 @@ namespace System.Threading
         ~Thread()
         {
 #if TARGET_UNIX || TARGET_BROWSER
-            WaitSubsystem.OnThreadExiting(this);
+            _waitInfo?.OnThreadExiting();
 #endif
             FreeInternal();
         }

